api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerMapData.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * This file is part of packetevents - https://github.com/retrooper/packetevents
+ * Copyright (C) 2024 retrooper and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package shadow.utils.holders.packet.custom;
+
+import com.github.retrooper.packetevents.event.PacketSendEvent;
+import com.github.retrooper.packetevents.manager.server.ServerVersion;
+import com.github.retrooper.packetevents.protocol.packettype.PacketType;
+import com.github.retrooper.packetevents.wrapper.PacketWrapper;
+import net.kyori.adventure.text.Component;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class WrapperPlayServerMapData extends PacketWrapper<WrapperPlayServerMapData> {
+
+    private int mapId;
+    private byte scale;
+    private boolean locked;
+    private MapIcon[] icons;
+    private int columns, rows, x, z;
+    private byte[] bytes;
+
+    public WrapperPlayServerMapData(PacketSendEvent event) {
+        super(event);
+    }
+
+    //The only change occurred here, where the array's length became optional
+    //https://wiki.vg/index.php?title=Protocol&oldid=17753#Map_Data 1.19
+    //https://wiki.vg/index.php?title=Protocol&oldid=18067#Map_Data 1.19.2
+
+    //hasIcons can also be called trackingPositions, but both ultimately do the same

```

Comment:
this is wrong

Reply:
Yep. I'll trust you'll handle the appriopriate changes, since you seem more acquainted with the version changes of this packet